### ISSUE TITLE
More generic support for SQL Server

### ISSUE THREAD

**Issue by andrewdmay** [OpenGamma/ElSql#12]
Is there anything in the SqlServer2008 configuration that is known to be specific to the 2008 version of SQL Server?
If not, it would be preferable to have a generic SqlServer config, and SqlServer2008 could be an alias for that (for anyone who is already using it).

I'm willing to put in a pull request if it sounds like a reasonable change.
Thanks,
Andrew

**Comment by jodastephen** [comment]
I believe that there is something in there that doesn't work on earlier versions, but it should be OK for later ones. An alias with suitable caveats sounds reasonable.

**Comment by andrewdmay** [comment]
Any idea what it is that's incompatible with earlier versions of SQL Server - we have to support SQL Server 2005 (but typically develop against 2008 or later). We're not currently using the paging functionality, but it might be useful for later as we move more of the codebase to ElSql.

**Comment by jodastephen** [comment]
No idea I'm afraid. If you can test the various bits of SQL on 2005 that would be great as we can then document the issues.

**Comment by jimmoores** [comment]
I think it was the new date and time types.

**Comment by andrewdmay** [comment]
We're going to try and spin up a test SQL Server 2005 instance so that we can test the paging SQL.
When you mention date and time types, is that elsewhere in OpenGamma? We are only using the ElSql module, so we only need to determine if the paging works in older versions of SQL Server.

**Comment by jodastephen** [comment]
ElSql doesn't know about date and time types so that must have been OpenGamma. So it might be that the OpenGamma restriction caused an unecessary restriction in ElSql.

Basically, the database specific stuff is defined in the [config class](https://github.com/OpenGamma/ElSql/blob/master/src/main/java/com/opengamma/elsql/ElSqlConfig.java). Providing these elements work in SQL server 2005 you should be fine. The rest of ElSql is standard SQL or just passing your SQL around.

Looking forward to hearing your results!

**Comment by jimmoores** [comment]
Yes, actually, the date and times thing must have been a deeper OG code restriction.  From what I can tell from looking at the code, the 2008 pagination should work on 2005.  I guess your experiment will find out one way or the other definitively.
